### PR TITLE
JITX-6112/bom-set-design-board-rules

### PR DESCRIPTION
**Pull request by erwin-lau**
### Changes

* move up `set-bom-vendors` and fix the sequence of `set-current-design`, `set-board`, and `set-rules`.

### CODE REVIEW

File: main.stanza [hunk 1]
```diff
@@ -7,14 +7,12 @@ defpackage main :
   import ocdb/utils/generic-components
   import helpers
 
+; Initialization
+initalize-helpers()
```
Comment (PhilippeFerreiraDeSousa) [inline main.stanza:11]: Probably no doing anything as you set the current design later

File: helpers.stanza [hunk 2]
```diff
@@ -10,6 +10,23 @@ defpackage helpers :
...
+; setup-design
+; ============
+public defn setup-design (name:String, board:Board
+                          --
+                          rules:Rules|False = false
+                          vendors:Tuple<String|AuthorizedVendor>|False = false
+                          quantity:Int|False = false) :
+  set-current-design(name)
+  set-board(board)
+  set-rules $ match(rules:Rules) : rules
```
Comment (PhilippeFerreiraDeSousa) [inline helpers.stanza:23]: This won't compile, you meant `match(rules:Rules) : set-rules(rules)` ?

File: helpers.stanza [hunk 3]
```diff
@@ -10,6 +10,23 @@ defpackage helpers :
...
+  else : ocdb/utils/defaults/default-rules
+  set-bom-vendors $ match(vendors:Tuple<String|AuthorizedVendor>) : vendors
```
Comment (PhilippeFerreiraDeSousa) [inline helpers.stanza:25]: This does compile but this is a different behavior from what you want as set-bom-vendors(false) would overwrite the default.

File: helpers.stanza [hunk 4]
```diff
@@ -10,6 +10,23 @@ defpackage helpers :
...
+  else : ocdb/utils/design-vars/APPROVED-DISTRIBUTOR-LIST
+  set-bom-design-quantity $ match(quantity:Int) : quantity
```
Comment (PhilippeFerreiraDeSousa) [inline helpers.stanza:27]: This one won't compile

File: helpers.stanza (hunk 3 above)
Comment (erwin-lau) [reply helpers.stanza:25]: There is a `else :` part to cater for the `false` case. I have checked with a `println(bom-vendors())` in main.
I can rewrite to put the default value in the parameter list though

Comment (PhilippeFerreiraDeSousa) [reply helpers.stanza:25]: Oh, my bad. We don't have the same indentation convention

**Review by PhilippeFerreiraDeSousa — APPROVED** [review]

**Pull request merged**